docs/snippets/provider-aws-access.md

```diff
@@ -95,6 +95,94 @@ spec:
 
 **NOTE:** In case of a `ClusterSecretStore`, Be sure to provide `namespace` for `serviceAccountRef` with the namespace where the service account resides.
 
+## EKS Pod Identity Setup
+
+In order to use EKS Pod Identity Agent, create a role like this:
+
+```json
+{
+    "Statement": [
+        {
+            "Action": [
+                "secretsmanager:GetResourcePolicy",
+                "secretsmanager:GetSecretValue",
+                "secretsmanager:DescribeSecret",
+                "secretsmanager:ListSecretVersionIds"
+            ],
+            "Effect": "Allow",
+            "Resource": [
+                "*"
+            ]
+        }
+    ],
+    "Version": "2012-10-17"
+}
+```
+
+```json
+{
+    "Version": "2012-10-17",
+    "Statement": [
+        {
+            "Sid": "AllowEksAuthToAssumeRoleForPodIdentity",
+            "Effect": "Allow",
+            "Principal": {
+                "Service": "pods.eks.amazonaws.com"
+            },
+            "Action": [
+                "sts:AssumeRole",
+                "sts:TagSession"
+            ]
+        }
+    ]
+}
+
+```
+
+
+Install ESO using helm and define these values:
+
+```yaml
+serviceAccount:
+  annotations:
+    eks.amazonaws.com/role-arn: arn:aws:iam::111122223333:role/my-role
+  name: external-secrets
+```
+
+Create a pod association:
+
+```
+aws eks create-pod-identity-association --cluster-name my-cluster --role-arn arn:aws:iam::111122223333:role/my-role --namespace external-secrets --service-account external-secrets
+```
+
+Then create a secret store like this:
+
+```yaml
+apiVersion: external-secrets.io/v1beta1
+kind: SecretStore
+metadata:
+  name: store
+spec:
+  provider:
+    aws:
+      service: SecretsManager
+      region: eu-central-1
+      # do a sts:assumeRole before fetching secrets
+      role: arn:aws:iam::111122223333:role/my-role
```

Comment:
Thank you for writing this doc! I'm afraid I tried to use EKS Pod Identity by specifying `spec.provider.aws.role`. However, it didn't work because the pod-identity-assumed role failed to assume its own role. 

L148 indicates that the service account has the annotation for [IRSA](https://docs.aws.amazon.com/eks/latest/userguide/associate-service-account-role.html). I wonder if IRSA was used instead of Pod Identity for a proof of concept. What do you think?

Reply:
>I wonder if IRSA was used instead of Pod Identity for a proof of concept. What do you think?

I was doing a spike on Pod Identity a couple days ago and ended up testing this scenario. I thought Pod identity was working but I was actually just using IRSA :man_facepalming:.

Reply:
@devonhk thank you for your comment! I think an IAM role chaining would be required for EKS Pod Identity as https://github.com/external-secrets/external-secrets/issues/2951#issuecomment-1868060588 describes.

Reply:
is there anything I need to update? :D

Reply:
> is there anything I need to update? :D

I think it would be beneficial to remove the `role` here. ESO doesn't need to assume a role as it has permissions to read secrets already.

Reply:
@moolen So, what about the error note down below?
>unable to create session: an IAM role must be associated with service account ...

Do i remove that too since it's no longer assuming the role?

Reply:
The error is still valid and it's a good idea to mention it here. People will get this wrong! Though i'd call it something along the lines of: eso can not impersonate service accounts which have iam roles bound using pod identity.

Reply:
done